### PR TITLE
use sed to translate optional -> oneof

### LINKED ISSUE #4258: Support optional fields in proto
A [change proposed](https://github.com/open-telemetry/opentelemetry-proto/pull/336) in the opentelemetry-proto uses `optional` fields, which are not supported by the gogo proto library. A few options have been proposed on how to move forward:
1. fork the gogo library and add support for optional fields
2. replace the library with manual serialization

### PR DESCRIPTION
**Pull request by codeboten**
Using sed to substitute `optional` fields with `oneof` as suggested by @bogdandrutu: https://github.com/open-telemetry/opentelemetry-collector/issues/4258#issuecomment-1049379599

>@codeboten one option which I don't know if we thought/discuss about is to actually rely on oneof for the "optional/pointer":
>optional fixed64 foo = 1 -> oneof foo_ { fixed64 foo = 1;}

This is a better alternative than #4750 as it allows us to continue using the proto generation as before.

Fixes #4258

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
LGTM in general. I think this is a good compromise to move things forward with the optional, and unblock proto changes

**Comment by codeboten** [comment]
Output from sending min/max for histo/expohisto datapoints:

```
    metrics {
      name: "histogram"
      unit: "1"
      histogram {
        data_points {
          sum: 1234.0
          min: 10.0
          max: 694.1
        }
        aggregation_temporality: AGGREGATION_TEMPORALITY_CUMULATIVE
      }
    }
    metrics {
      name: "exponential-histogram"
      unit: "1"
      exponential_histogram {
        data_points {
          sum: 12345.0
          positive {
          }
          negative {
          }
          min: 105.0
          max: 694.15
        }
        aggregation_temporality: AGGREGATION_TEMPORALITY_DELTA
      }
    }
```

Will leave this in draft until https://github.com/open-telemetry/opentelemetry-proto/pull/279 is merged

**Comment by codeboten** [comment]
@tigrannajaryan please take a look at this draft as you've reviewed the PR this replaces.

File: model/pdata/generated_metrics.go [hunk 1]
```diff
@@ -1438,6 +1438,50 @@ func (ms HistogramDataPoint) SetFlags(v MetricDataPointFlags) {
...
+}
+
+// Min returns the min associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) Min() float64 {
+	return (*ms.orig).GetMin()
+}
+
+// SetMin replaces the min associated with this HistogramDataPoint.
+func (ms HistogramDataPoint) SetMin(v float64) {
+	(*ms.orig).Min_ = &otlpmetrics.HistogramDataPoint_Min{
```
Comment (tigrannajaryan) [inline model/pdata/generated_metrics.go:1458]: Nit: this can check to see if `(*ms.orig).Min_ != nil` and use it without allocating a new one.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Comment by codeboten** [comment]
Closing this, changes will be incorporated into #5064

**Pull request closed without merging**